x-pack/solutions/security/plugins/security_solution/public/detections/components/alerts_table/index.tsx

```diff
@@ -313,13 +318,23 @@ export const AlertsTableComponent: FC<Omit<DetectionEngineAlertTableProps, 'serv
         inspect: null,
       });
     },
-    [dispatch, onLoad, setQuery, tableType]
+    [dispatch, onLoad, setQuery, toggleColumnRef, tableType]
   );
   const userProfiles = useFetchUserProfilesFromAlerts({
     alerts: tableContext?.alerts ?? [],
     columns: tableContext?.columns ?? [],
   });
 
+  useEffect(
+    () => () => {
+      if (toggleColumnRef?.current) {
+        // Unset the toggle column action on unmount
+        toggleColumnRef.current = null;
+      }
+    },
+    [toggleColumnRef]
```

Comment:
`toggleColumnRef` should not be required in dependency array, correct?

Reply:
+1

Reply:
It's one of those cases where it seems redundant but ESLint complains. It shouldn't change between renders though

<img width="1041" alt="image" src="https://github.com/user-attachments/assets/039a26ef-96f0-4143-9338-2963690b434b" />

Reply:
Interesting, it seems it does not understand that it is a `ref`.

Reply:
I removed the effect (see https://github.com/elastic/kibana/pull/208052#issuecomment-2615246820)